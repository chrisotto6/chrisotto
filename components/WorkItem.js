import kebabCase from '@/lib/utils/kebabCase'

const WorkItem = ({ company, role }) => {
  return (
    <div data-testid={`job-${kebabCase(company)}`}>
      <h3>{company}</h3>
      {role.map((role) => (
        <div key={role.title}>
          <h4>{role.title}</h4>
          <span>{`${role.monthYearStart} - ${role.monthYearEnd}`}</span>
          <ul className="list-outside leading-snug">
            {role.jobDescription.map((desc) => (
              <li key={desc}>{desc}</li>
            ))}
          </ul>
        </div>
      ))}
    </div>
  )
}

export default WorkItem
