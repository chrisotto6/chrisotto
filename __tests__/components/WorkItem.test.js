import { render } from '@testing-library/react'
import WorkItem from '@/components/WorkItem'

it('renders WorkItem correctly', () => {
  const job = {
    company: 'Test LLC',
    role: [
      {
        title: 'Senior Coder',
        monthYearStart: 'Aug 2022',
        monthYearEnd: 'Current',
        jobDescription: ['I did this', 'And this', 'And then lastly this'],
      },
    ],
  }
  const { container } = render(<WorkItem company={job.company} role={job.role} />)
  expect(container).toMatchSnapshot()
})
