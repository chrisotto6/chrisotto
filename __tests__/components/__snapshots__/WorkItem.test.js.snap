// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WorkItem correctly 1`] = `
<div>
  <div
    data-testid="job-test-llc"
  >
    <h3>
      Test LLC
    </h3>
    <div>
      <h4>
        Senior Coder
      </h4>
      <span>
        Aug 2022 - Current
      </span>
      <ul
        class="list-outside leading-snug"
      >
        <li>
          I did this
        </li>
        <li>
          And this
        </li>
        <li>
          And then lastly this
        </li>
      </ul>
    </div>
  </div>
</div>
`;
